src/textual/widgets/_select.py

```diff
@@ -385,3 +385,8 @@ def action_show_overlay(self) -> None:
         select_current = self.query_one(SelectCurrent)
         select_current.has_value = True
         self.expanded = True
+        
+    def clear(self) -> None:
+        """Clear the selected option."""
+        self.value = None
+        self.query_one(SelectCurrent).update(None)
```

Comment:
If you look at `_watch_value`, you will see that the watcher method will already update the `SelectCurrent`, so `self.value = None` is sufficient.

Also, make sure to include an empty newline at the end of the file.
(If you install the pre commit hooks, they'll help you.)